install/windows/build-wxs.py

```diff
@@ -123,122 +84,160 @@ def main():
     with open(version_filename) as f:
         # Read the Dangerzone version from share/version.txt, and remove any potential
         # -rc markers.
-        version = f.read().strip().split("-")[0]
+        dangerzone_version = f.read().strip().split("-")[0]
 
-    dist_dir = os.path.join(
+    dangerzone_product_upgrade_code = "12B9695C-965B-4BE0-BC33-21274E809576"
+
+    build_dir = os.path.join(
         os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))),
         "build",
-        "exe.win-amd64-3.12",
     )
+
+    cx_freeze_dir = "exe.win-amd64-3.12"
+
+    dist_dir = os.path.join(build_dir, cx_freeze_dir)
+
     if not os.path.exists(dist_dir):
         print("You must build the dangerzone binary before running this")
         return
 
-    data = {
-        "id": "TARGETDIR",
-        "name": "SourceDir",
-        "dirs": [
-            {
-                "id": "ProgramFilesFolder",
-                "dirs": [],
-            },
-            {
-                "id": "ProgramMenuFolder",
-                "dirs": [],
-            },
-        ],
-    }
+    # Prepare data for WiX file harvesting from the output of cx_Freeze
+    data = build_data(
+        dist_dir,
+        cx_freeze_dir,
+        "INSTALLFOLDER",
+        "Dangerzone",
+    )
 
-    data["dirs"][0]["dirs"].append(
-        build_data(
-            dist_dir,
-            "exe.win-amd64-3.12",
-            "INSTALLDIR",
-            "Dangerzone",
-        )
+    # Add the Wix root element
+    wix_el = ET.Element(
+        "Wix",
+        {
+            "xmlns": "http://wixtoolset.org/schemas/v4/wxs",
```

Comment:
Just wanted to say that while this is not supposed to be a link, it redirects you to Smashing Pumpkin's "Bullet with Butterfly Wings", and I'm down with it.